api/src/task/opportunities/export_opportunity_data_task.py

```diff
@@ -0,0 +1,108 @@
+import json
+import logging
+from enum import StrEnum
+from typing import Iterator, Sequence
+
+from smart_open import open
+from sqlalchemy import select
+from sqlalchemy.orm import noload, selectinload
+
+import src.adapters.db as db
+import src.adapters.db.flask_db as flask_db
+from src.api.opportunities_v1.opportunity_schemas import OpportunityV1Schema
+from src.db.models.opportunity_models import CurrentOpportunitySummary, Opportunity
+from src.services.opportunities_v1.opportunity_to_csv import opportunities_to_csv
+from src.task.task import Task
+from src.task.task_blueprint import task_blueprint
+from src.util.datetime_util import get_now_us_eastern_datetime
+
+logger = logging.getLogger(__name__)
+
+
+@task_blueprint.cli.command(
+    "export-opportunity-data",
+    help="Generate JSON and CSV files containing an export of all opportunity data",
+)
+@flask_db.with_db_session()
+def export_opportunity_data(db_session: db.Session) -> None:
+    ExportOpportunityDataTask(db_session).run()
+
+
+class ExportOpportunityDataTask(Task):
+    class Metrics(StrEnum):
+        RECORDS_LOADED = "records_loaded"
+
+    def __init__(
+        self,
+        db_session: db.Session,
+    ) -> None:
+        super().__init__(db_session)
+
+        FILE_NAME: str = "opportunity_data"
+        self.current_timestamp = get_now_us_eastern_datetime().strftime("%Y-%m-%d_%H-%M-%S")
+
+        # Surely there is a better way to do paths in python?
+        # I tried pathlib's Path.cwd() thinking it would resolve
+        # to /api/src/task/opportunities/ but it resolved to just /api
+        self.FILE_PATH = "/api/src/task/opportunities/output/"
```

Comment:
For the path, don't specify it in this class at all. Instead have it load an env var config - that way the path is fairly configurable (unit tests can put in some temp directory, non-local can put an S3 path).

You can see this sort of pattern for other string env vars in some of the other tasks like https://github.com/navapbc/simpler-grants-gov/blob/main/api/src/search/backend/load_opportunities_to_index.py#L21

Reply:
This should be implemented with the latest commit